test/modules/aws-security-group-integration.ts

```diff
@@ -532,6 +534,103 @@ describe('Security Group Integration Testing', () => {
 
   it('applies the vpc removal', apply());
 
+  // tests cycle in security rules
+  it(
+    'adds a new security group A',
+    query(`  
+    INSERT INTO security_group (description, group_name)
+    VALUES ('Security Group Test A', '${prefix}sgtestA'), ('Security Group Test B', '${prefix}sgtestB');
+  `),
+  );
+
+  it(
+    'adds security group rules pointing to the other ones (A)',
+    query(`
+  INSERT INTO security_group_rule(description, security_group_id, source_security_group, is_egress) 
+  VALUES ('${prefix}sgtestA', (SELECT id FROM security_group WHERE group_name='${prefix}sgtestA'),
+  (SELECT id FROM security_group WHERE group_name='${prefix}sgtestB'), false);
+  `),
+  );
+  it(
+    'adds security group rules pointing to the other ones (B)',
+    query(`
+  INSERT INTO security_group_rule(description, security_group_id, source_security_group, is_egress) 
+  VALUES ('${prefix}sgtestB', (SELECT id FROM security_group WHERE group_name='${prefix}sgtestB'),
+  (SELECT id FROM security_group WHERE group_name='${prefix}sgtestA'), false);
+  `),
+  );
```

Comment:
Also, this could be condensed to just

```suggestion
  it(
    'adds self-referential security groups A and B',
    query(`  
    INSERT INTO security_group (description, group_name)
    VALUES ('Security Group Test A', '${prefix}sgtestA'), ('Security Group Test B', '${prefix}sgtestB');
    
    INSERT INTO security_group_rule(description, security_group_id, source_security_group, is_egress) 
  VALUES ('${prefix}sgtestA', (SELECT id FROM security_group WHERE group_name='${prefix}sgtestA'),
  (SELECT id FROM security_group WHERE group_name='${prefix}sgtestB'), false);
  
  INSERT INTO security_group_rule(description, security_group_id, source_security_group, is_egress) 
  VALUES ('${prefix}sgtestB', (SELECT id FROM security_group WHERE group_name='${prefix}sgtestB'),
  (SELECT id FROM security_group WHERE group_name='${prefix}sgtestA'), false);
  `),
  );
```